packages/cmrjs/src/echo10Modifiers.ts

```diff
@@ -45,8 +45,38 @@ export function updateEcho10XMLGranuleUrAndGranuleIdentifier({
 
   moddedXml.Granule ??= {};
   moddedXml.Granule.GranuleUR = granuleUr;
-  moddedXml.Granule.DataGranule ??= {};
-  moddedXml.Granule.DataGranule.ProducerGranuleId = producerGranuleId;
+
+  if (!moddedXml.Granule.DataGranule) {
+    moddedXml.Granule.DataGranule = {};
+  }
+
+  const dataGranule = moddedXml.Granule.DataGranule as any;
+  const orderedDataGranule = new Map<string, any>();
+
+  // ECHO10 DataGranule element order as defined in the XSD schema
+  // https://git.earthdata.nasa.gov/projects/EMFD/repos/echo-schemas/browse/schemas/10.0/Granule.xsd
```

Comment:
If the schema changes, such as adding an element in DataGranule, then the changes here will drop that element.

Reply:
Can we add the following, so that any extra element in the original xml will be appended?
for (const key of Object.keys(dataGranule)) {
    if (!schemaOrder.includes(key)) {
      orderedDataGranule[key] = dataGranule[key];
    }
  }

Reply:
I think we could, however the problem with that approach is it probably won't be in the correct order - if additional keys are added the reading library doesn't ensure order, *and* the above approach assumes that the sequence will append new key values to the end of the sequence which also isn't promised. 

If we really need to do this, I think we need to take on the extra work to refactor our usage of xml2js or use a different parser.

Reply:
So - to be clear, you're advocating for the additional work of re-write the read logic to use preserveChildrenOrder or a different XML parser prior to this patch?

Reply:
Addressed PR feedback re^ failing on extra keys 

b595385efc0

Thanks for the review @jennyhliu  🙇🏻